api/v1alpha1/clusterconfig_types.go

```diff
@@ -25,6 +25,9 @@ type ClusterConfigSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
+	// +optional
+	ImageRepository *ImageRepository `json:"imageRepository,omitempty"`
```

Comment:
```suggestion
	// +optional
	KubernetesImageRepository *ImageRepository `json:"kubernetesImageRepository,omitempty"`
```

Is the image repository some known term in CAPI context? If this value will affect only some images the name could be more specific.

Reply:
Agreed, I would prefer to make this name more explicit naming, including changing it from `Repository` to `Registry`.

Reply:
👍 I used `imageRepository` because thats what the kubeadm config has but it the benefit of the context being part of `kubeadmConfigSpec/clusterConfiguration` struct. I'll make it more explicit here.

Reply:
@jimmidyson the `Repository` vs `Registry` doesn't bother me too much, I can see why in kubeadm it is called `repository`, but I think "registry" is more commonly used in Kubernetes docs, I'll change it too.